roles/webserver/tasks/main.yml

```diff
@@ -27,7 +27,7 @@
     group: "{{ unicorn_user }}"
   become: yes
   tags: unicorn_config
-  notify: reload unicorn
+  notify: reload unicorn configs
```

Comment:
So this task actually doesn't need the USR2+QUIT, right? I guess it would be better to send HUP here. We could do something like this: `systemctl --singal HUP kill unicorn`.

Semantically, the HUP signal is more like a reload and the USR2+QUIT is like a restart but we can't tell systemd that. We can just choose one of them to be the `ExecReload` and need to handle the other one with a kill signal.

What do you think, should we switch it around?

Reply:
This will call the same (updated) reload action on the service, which is now `USR2`. This separate handler is run if/when we change the `unicorn.rb` config file during provisioning. I'm pretty sure the new reload will pick up these config changes as expected, as I understand it, it replaces the master process and all the workers with new ones, just in a smoother way.

Reply:
The name change here was to avoid clashes, as we actually use some slightly different logic surrounding the two handlers in provision and deploy...

Reply:
No objection to leaving this as is. It should work. My point was that we could be more efficient by sending HUP because it will only reload the configs without replacing the executable and reloading all gems. But introducing another way of reloading is also introducing complexity. So I'm happy to leave it as is.